common/flatpak-dir.c

```diff
@@ -198,6 +198,201 @@ get_config_dir_location (void)
   return (const char *)path;
 }
 
+static FlatpakRemoteState *
+flatpak_remote_state_new (void)
+{
```

Comment:
I don't think this really make a difference. In the end as we're reusing the RemoteState in multiple places it turns out that we don't really know ahead of time what will be required for all the uses of a state, so most everything is now using the "optional" version, and then ensuring what they need is there and erroring out if not.

Reply:
Personally, I find the main difference is that you would always have a complete object which you can then treat as effectively immutable. But its a more of a stylistic preference and up to you.

Reply:
Actually, this will not really work because of the way _flatpak_dir_get_remote_state() calls _flatpak_dir_fetch_remote_state_metadata_branch which calls flatpak_dir_pull which takes a FlatpakRemoteState.